language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs

```diff
@@ -24,23 +24,29 @@ main = do
     damlc <- locateRunfiles (mainWorkspace </> "compiler" </> "damlc" </> exe "damlc")
     daml2ts <- locateRunfiles (mainWorkspace </> "language-support" </> "ts" </> "codegen" </> exe "daml2ts")
     davl <- locateRunfiles ("davl" </> "released")
-    yarnPath : args <- getArgs
+    yarnPath : damlTypes : args <- getArgs
     yarn <- locateRunfiles (mainWorkspace </> yarnPath)
     withTempDir $ \rootDir ->
       withArgs args (
         defaultMain $
           withResource
-          (yarnInstall yarn rootDir) (\_ -> pure ())
+          (yarnInstall yarn damlTypes rootDir) (\_ -> pure ())
           (\_ -> tests rootDir yarn damlc daml2ts davl)
         )
 
-yarnInstall :: FilePath -> FilePath -> IO ()
-yarnInstall yarn rootDir = do
+yarnInstall :: FilePath -> FilePath -> FilePath -> IO ()
+yarnInstall yarn damlTypes rootDir = do
   let here = rootDir </> "pre-test"
   let dummyTs = here </> "dummy-ts"
   createDirectoryIfMissing True dummyTs
+  copyDirectory damlTypes (rootDir </> "daml-types")
   writePackageConfigs dummyTs
   withCurrentDirectory rootDir $ yarnProject' yarn ["install"]
+  where
+    copyDirectory from to = do
```

Comment:
We already have `copyDirectory` in `DA.Daml.Project.Util` which contrary this function should work properly if the directory structure is nested (presumably that’s not the case now but that’s easy to change accidentally). `daml-project-config` is a bit of a weird place for it to live so feel free to move it to `da-hs-base` or something.

Reply:
@shayne-fletcher Can you please take care of that in a follow-up PR?

Reply:
@hurryabit Sure thing.